### PR TITLE
Add CircleCI job

### PR DESCRIPTION
**Pull request by bersace**
Hi,

I suggest to add regression tests executed on every contributions in a reproducable tests. With this patch, we have basic infrastructure to execute pgbadger : perl interpreter, code checkout, GitHub integration.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+    - run:
+        name: Install git and ssh for checkout
+        command: apt update -y && apt install -y git ssh
+    - checkout
+    - run:
+        name: Testing perl syntax
+        command: |
+          perl -c pgbadger
+          ./pgbadger --help
```
Comment (bersace) [inline .circleci/config.yml]: @jpargudo Here we execute shell commands to test pgbadger. For now this is a minimal sanity check. We can add real use cases later : csvlog, xz compression, exclude query, etc.

**Comment by darold** [comment]
I would prefer a simple TAP based regression tests. And it is not clear for me that CircleCI is a full open source tool, can you confirm?

**Comment by daamien** [comment]
ims que Circle CI est open source : https://github.com/circleci

L'avantage d'avoir un système intégré à github c'est que ça aide les contributeurs qui veulent proposer un patch.

**Comment by darold** [comment]
I gave you my advice, use Tap. But I'm no longer the maintainer of pgBadger so feel free to use what you want and what you think is better.

**Comment by bersace** [comment]
@darold i guess that TAP is usable from bash ? By TAP, you mean https://testanything.org/ ? I didn't know this.

@darold CircleCI is more open source than using GitHub :-) Our test case is rather simple. It's easy to move to e.g. travis or a custom CI if ever we feel trapped in CircleCI.

Let's iterate on this later. Thanks for the review.

**Comment by darold** [comment]
Yes I was talking about this Tap, which is also used in PostgreSQL regression tests but more generally used with most Perl projects. The main advantage is that is doesn't depend on online or commercial services and as you see it can be run offline from a shell. But as I said, feel free to use what is best for you, I don't want to influence the next decisions about pgBadger.

**Comment by bersace** [comment]
@darold Thanks. Actually, the purpose of CircleCI is to run TAP tests (with e.g. runprove or anything else) on each PR to validate contributions automatically.

**Comment by darold** [comment]
Ok, seems fine.

**Pull request merged**